helpers.js

```diff
@@ -29,6 +31,38 @@ module.exports.isCustomCredentials = (settings) => {
     settings.previewToken !== previewToken
 }
 
+function cleanupQueryParameters (query) {
+  const cleanQuery = Object.assign({}, query)
+  delete cleanQuery.space_id
+  delete cleanQuery.delivery_token
+  delete cleanQuery.preview_token
+  delete cleanQuery.reset
+  return cleanQuery
+}
+
+function updateSettingsQuery (request, response, settings) {
+  const cleanQuery = cleanupQueryParameters(request.query)
+
+  let settingsQuery = Object.assign({}, cleanQuery, {
+    editorial_features: settings.editorialFeatures ? 'enabled' : 'disabled'
+  })
+
+  if (isCustomCredentials(settings)) {
+    settingsQuery = Object.assign(settingsQuery, {
+      space_id: settings.spaceId,
+      delivery_token: settings.deliveryToken,
+      preview_token: settings.previewToken
+    })
+  }
```

Comment:
So a little confused about the above two functions:
- the `cleanupQueryParameters` function seemingly strips a query of everything...why not just initialize an empty query?
- Then, we take a newly created query and populate the settings with what was in already stored in our session. Is my understanding correct? The spec says we should populate the fields on the settings screen with the incorrect credentials (but of course showing the validation errors to the user, which i'll assume is done).

Reply:
The function keeps api & locale setting in the query. Thats why an empty object is not a solution.

Plus, our session storage is independent form the query. We will try to apply the credentials from the query, store them in the session when they are valid, but never keep the credentials in the query.